src/__tests__/recall-by-number.test.ts

```diff
@@ -0,0 +1,47 @@
+import { describe, expect, test } from '@jest/globals'
+import { parsePresetVariableOption } from '../recall-by-number.js'
+import { CompanionOptionValues } from '@companion-module/base'
+import { PresetRecall } from '../camera/commands.js'
+import { prettyBytes } from '../visca/utils.js'
+import { MockContext } from './mock-context.js'
+
+describe('test recall preset values', () => {
+	test('Variable value contant used dec 37', async () => {
+		const context = new MockContext()
+		const input: CompanionOptionValues = { val: '37' }
+		const result = await parsePresetVariableOption(input, context)
+		expect(typeof result).toBe('object')
```

Comment:
Because `typeof null === 'object'`, I'm always leery of any comparison against `'object'`.  Moreover you're fighting the TypeScript type system in the very next line.  If you wrote slightly more idiomatic TypeScript that moved the type-check into control flow, then `result` would be automatically understood as a `CompanionOptionValues` in the next line.  I think you should do this as:

```ts
if (typeof result === 'string') {
    expect(typeof result !== 'string').toBe(true)
} else {
    const commandBytes = PresetRecall.toBytes(result)
    expect(prettyBytes(commandBytes)).toBe('[81 01 04 3F 02 25 FF]')
}
```

and similar for the other tests.

Reply:
@jswalden  with your comment of refactoring the creation of the options into a function, what do you think about putting the verification into a function?  On one hand I like the idea for tidiness, but then errors come out of that function vs directly in the test and not sure if that clouds readability.  Welcome any thoughts you might have on that.